### PR TITLE
Fix misleading mcq behavior - TNL-1955

### PR DESCRIPTION
**Pull request by mushtaqak**
On MCQs questions with max attempts more than one, if a user attempts a problem wrong, it is marked wrong with a cross 'X' sign. 

Then user saves other choice and goes to other pages and comes back to the problem earlier, the 'X' sign is still there next to a saved choice which should not be a case.

Actually when a user answers a problem wrong, the state correctness property is changed to incorrect but when a user saves a choice, the state correctness property is not changed.

Thus when user re-visits the page, incorrect value is fetched ( which was not changed). So we need to change state correctness property on problem_save and make it to default value.

### CODE REVIEW

**Comment by waheedahmed** [comment]
@symbolist FYI.

File: common/lib/xmodule/xmodule/tests/test_capa_module.py [hunk 1]
```diff
@@ -931,6 +931,39 @@ def test_save_problem(self):
...
+        module_correctness = module.correct_map[answer_id].get('correctness', None)
+        self.assertEqual(state_correctness, module_correctness)
+        self.assertEqual(state_correctness, 'incorrect')
+
+        # Save a choice
+        get_request_dict = {CapaFactory.input_key(): 'choice_3'}
+        result = module.save_problem(get_request_dict)
+
+        # Expect that the result is success
+        self.assertTrue('success' in result and result['success'])
```
Comment (adampalay) [inline common/lib/xmodule/xmodule/tests/test_capa_module.py:959]: what about `self.assertIsNotNone(result.get('success'))`

**Comment by adampalay** [comment]
@mushtaqak does this mean that if I answer a problem incorrectly, then I return to the problem and click save, it no longer store the information that I had answered the question incorrectly? Rather than changing the student state, would it make more sense to just add a piece of logic that determines if we should display the X or not?

**Comment by adampalay** [comment]
Like, if you've answered a problem incorrectly, but then saved a new answer, you shouldn't display the red x

**Pull request closed without merging**